### DISCUSSION TITLE
Welcome to sessionless Discussions!

### DISCUSSION THREAD

**Discussion by zach-freyja** [planet-nine-app/sessionless#20] [Announcements]
Standardizing naming, and message structure:

So since each server implementation might get requests from clients in other languages, we need to standardize naming of the shared properties. Those are:

* `user uuid`
* `time stamp`
* `public key`

I went with the JavaScript way of doing the three (camelCase) because of web being such a big use case. So those would become:

* `userUUID`
* `timestamp`
* `publicKey`

I like this because though not all languages use camelCase, I feel it won't be unfamiliar to anyone implementing Sessionless.

I think the other reasonable option is snake_case:

* `user_uuid`
* `timestamp`
* `public_key`

I don't like `userUuid` for totally nitpicky reasons.

**Comment by Galacticai** [comment]
#### public key:

I suggest staying away from language keywords such as public and private and keeping the `publicKey` format 

Example: c# won't allow using public/private as names.. But it works if following c# naming rules (PascalCase) like "Public".. However it still feels wrong imo even tho it works

#### user uuid:

userUUID is more expressive and intuitive than or uuid alone

also userUuid just looks weird.. Some things look ok like this but UUID shall stay UUID 👽

####  time stamp:
timestamp in my mind usually refers to unix time (int64)
but it could also be timeMS.. Or just time since the value is clearly a number which indicates time is a unix time stamp

---

As for snake_case... I personally hate having to click shift to type _ ... And it just doesn't look pleasing imo
I usually only use underscore when something needs extra info in its name.. But even then, I still dont do full snake case

---

At least that's what I think

**Reply by zach-freyja** [reply]
agreed on all points. I actually didn't mean to suggest the ones with spaces, I was trying to just list them out, but I probably shouldn't have formatted them as code. So just to be clear, I think we're both aligned on:

* `userUUID`
* `publicKey`
* `timestamp`

I'm no fan of snake case either, but open to hearing what others think.

**Comment by FssAy** [comment]
If we're taking the language specifications into the account Rust uses snake_case, but there shouldn't be an issue with any naming conventions as it's possible to manually specify it for the serialization.
Example from the Rust implementation:

```rs
#[serde(rename_all="camelCase")]
pub struct PayloadRegister {
    pub public_key: String,
    pub timestamp: String,
    pub entered_text: String,
}
```

As for the `userUUID` I think that using `uuid` doesn't remove any clarity and overall looks better. UUIDs in this scenario are used only to identify a specific user, so it might be unnecessary to explicitly include the "user" prefix. I believe these names should be very short, but also understandable to the server and implementer alike without the need to be self explanatory to third parties that have no impact on the connection.

That's why I suggest:
- `uuid`
- `ts` / `timestamp`
- `pubKey`

­
_But that's just my tastes and I wanted to contribute a bit with this perspective._

**Reply by zach-freyja** [reply]
You know, I agree with all your points. And for `userUUID` leaving off user makes it so we don't have to define what a "user" is, which is nice since servers, bots, and other things might be authenticated. I like `timestamp` better than `ts` just thinking about people needing to translate the concept to languages other than English. `pubKey` is probably more widely used than `publicKey` so I'm good with that too.

* `uuid`
* `timestamp`
* `pubKey`

Would be my vote.

**Reply by FssAy** [reply]
I also agree on the `timestamp` point, `ts` is too abstract.
